api/src/main/java/com/epam/pipeline/manager/cluster/KubernetesManager.java

```diff
@@ -717,4 +712,57 @@ public boolean isNodeAvailable(final Node node) {
     public boolean isNodeUnavailable(final Node node) {
         return !isNodeAvailable(node);
     }
+
+    public Service createService(final String serviceName, final Map<String, String> runIdMap,
+                                 final List<ServicePort> ports) {
+        try (KubernetesClient client = getKubernetesClient()) {
+            final Service service = client.services().createNew()
+                    .withNewMetadata()
+                    .withName(serviceName)
+                    .withNamespace(kubeNamespace)
+                    .withLabels(runIdMap)
+                    .endMetadata()
+                    .withNewSpec()
+                    .withPorts(ports)
+                    .withSelector(runIdMap)
+                    .endSpec()
+                    .done();
+            Assert.notNull(service, messageHelper.getMessage(MessageConstants.ERROR_KUBE_SERVICE_CREATE, serviceName));
+            return service;
+        }
+    }
+
+    public Service getService(final String labelName, final String labelValue) {
+        try (KubernetesClient client = getKubernetesClient()) {
+            final Service service = findServiceByLabel(client, labelName, labelValue);
+            if (Objects.isNull(service)) {
+                return null;
+            }
+            return service;
+        }
+    }
+
+    public boolean deleteService(final Service service) {
+        try (KubernetesClient client = getKubernetesClient()) {
+            final Boolean deleted = client.services().delete(service);
+            if (Objects.isNull(deleted) || !deleted) {
+                LOGGER.debug("Failed to delete service '{}'", service.getMetadata().getName());
+            }
+            return deleted;
+        }
+    }
+
+    private Service findServiceByLabel(final KubernetesClient client, final String labelName, final String labelValue) {
```

Comment:
Let's return `Optional` from this method to handle nulls in a more graceful way.